src/Engine/Graphics/Indoor.cpp

```diff
@@ -1706,8 +1721,16 @@ void BLV_ProcessPartyActions() {  // could this be combined with odm process act
     }
 
     Vec3f oldPos = pParty->pos;
+    Vec3f savedspeed = pParty->speed;
 
+    // horizontal
+    pParty->speed.z = 0;
     ProcessPartyCollisionsBLV(sectorId, min_party_move_delta_sqr, &faceId, &faceEvent);
+    // vertical -  only when horizonal motion hasnt caused height gain
```

Comment:
Splitting the collision cycles in two, handling horizontal and vertical separately simplifies the response. We need to track less states this way. It meant we could drop tracking height gain modifications via `partySlopeMod` and associated space launch effects for example. In the typical case where we hit a vertical face in the horizontal movement section, the vertical will drop us down in one step. Consider previous behavior where you could jump halfway up a wall and as long as you kept moving towards it, it would count as a collision and drop you down very slowly. Now the first cycle handles collision with the wall and sliding along it, the second v cycle will experience no collision and drop or raise party appropriately. And similarly if you were already colliding with a wall and then tried to jump you often wouldn't.